### DISCUSSION TITLE
FreeRDP on Mac

### DISCUSSION THREAD

**Discussion by dbi1** [FreeRDP/FreeRDP#8578] [General]
Hello,

I am trying out FreeRDP on Mac, and noticed two things:

1) Is Quartz required for FreeRDP on Mac? When I simply type `xfreerdp` in the command line on Mac, Quartz shows up in the dock.

2) In another discussion I read that "xfreerdp on mac is not working well due to lacking X11 on mac" [here](https://github.com/FreeRDP/FreeRDP/discussions/7956). Was this a general comment on FreeRDP on Mac, or only related to the specific RemoteApp topic being discussed?

Thank you for any thoughts.

**Comment by akallabeth** [comment]
1. `xfreerdp` is a `X11` application, so yes, you need a `X11` server
2. there is also a native mac client available. (not as mature as `xfreerdp` but uses native drawing API)
3. if you want remote app you better look to other `FreeRDP` based clients on mac

**Reply by dbi1** [reply]
Thank you @akallabeth.

I am also interested in running FreeRDP as a server on Mac. Most of the documentation / help I was able to find seemed about how to run as a client.

Any pointer on where I might find a bit of documentation / help on running FreeRDP as a server on Mac? Or maybe a simple command line example?

Thank you again!

**Reply by akallabeth** [reply]
@dbi1 there is currently no frontend for the server parts that works on mac.
there is github.com/ogon-project/ and https://gitlab.gnome.org/GNOME/gnome-remote-desktop for linux (as well as the shadow-server from this repo) but there are no glue code implementations so far for other systems.

**Reply by dbi1** [reply]
Got it, thank you for explaining and for the quick answers, @akallabeth!

**Reply by AlexanderProd** [reply]
> there is also a native mac client available. (not as mature as xfreerdp but uses native drawing API)

Where can I find the native mac client? Unfortunately I couldn't find any. 

Edit: 
found it, I think it is [this](https://thincast.com/en) one.

**Reply by akallabeth** [reply]
@AlexanderProd no, that is a different client (although `FreeRDP` based)
* the native mac client is part of the FreeRDP project see https://github.com/FreeRDP/FreeRDP/tree/master/client/Mac
* there are no binaries, you need to build it with https://github.com/FreeRDP/FreeRDP/blob/master/scripts/bundle-mac-os.sh

**Reply by alex-eri** [reply]
> 3\. `FreeRDP` based clients on mac

which are they?

**Reply by akallabeth** [reply]
> > 3. `FreeRDP` based clients on mac
> 
> which are they?

1. `sdl-freerdp` can be used natively on mac
2. there is a native mac client you can build (deprecated, but working)
3. there is `thincast-client`
4. there is `royal tsx`
5. ... these are only from top of my head

**Comment by smil2k** [comment]
brew install freerdp give you xfreerdp (which needs x), but also it gives you sdl-freerdp which is working.

**Comment by geiseri** [comment]
I can say the sdl-freerdp is really good on MacOS.  I would push the homebrew guys to use that instead of X.

**Reply by smil2k** [reply]
For me the login screen is kind of funny. Which arguments are you using to make it more compatible with cocoa/masos?

**Reply by piyushgarg** [reply]
@geiseri sdl-freerdp is better than using quartz, but still issues between the display of macos and rdp. By default the sdl window open up very tiny and with resize it becomes unstable, some apps are pixelated but others are ok.
